project-base/app/tests/FrontendApiBundle/Functional/Product/ProductList/ProductListLoggedCustomerTest.php

```diff
@@ -150,6 +150,30 @@ public function testAddProductCreatesNewListWhenNewUuidIsProvided(string $produc
         $this->assertSame([$productToAddId], array_column($data['products'], 'id'));
     }
 
+    /**
+     * @dataProvider \Tests\FrontendApiBundle\Functional\Product\ProductList\ProductListTypesDataProvider::getProductListTypes
+     * @param string $productListType
+     */
+    public function testAddProductCreatesNewListWithNewUuidWhenUuidOfAnonymousListIsProvided(
+        string $productListType,
+    ): void {
+        $anonymousProductListUuid = $productListType === ProductListTypeEnum::COMPARISON
+            ? ProductListDataFixture::PRODUCT_LIST_COMPARISON_NOT_LOGGED_CUSTOMER_UUID
+            : ProductListDataFixture::PRODUCT_LIST_WISHLIST_NOT_LOGGED_CUSTOMER_UUID;
```

Comment:
adding a new type of list would require changing this condition. What about rewriting it to `match`, so it will be easier to cover the new product list with the test?

Reply:
another great idea, thanks! 3267f6668f